CentralHub.Api/Services/LocalizationService.cs

```diff
@@ -1,70 +1,209 @@
+using System.Collections.Immutable;
+using System.Linq;
+using App.ScopedService;
+using CentralHub.Api.Dtos;
 using CentralHub.Api.Model;
 
 namespace CentralHub.Api.Services;
 
-public class LocalizationService : ILocalizationService
+public class LocalizationService : IScopedProcessingService
 {
-    protected Thread? MeasurementRemover;
-    public TimeSpan MaxAge { get; set; }
+    private static readonly TimeSpan _sleepTime = TimeSpan.FromMinutes(2);
+    private readonly ILogger<LocalizationService> _logger;
 
-    private readonly List<ThreadState> validStates = new List<ThreadState>() { ThreadState.Running, ThreadState.WaitSleepJoin };
-    protected Dictionary<int, List<MeasurementGroup>> _measurements = new Dictionary<int, List<MeasurementGroup>>();
+    private readonly IAggregatedMeasurementRepository _aggregatorRepository;
 
-    public void AddMeasurements(int id, IReadOnlyCollection<Measurement> measurements)
+    private readonly ITrackerRepository _trackerRepository;
+
+    private readonly IRoomRepository _roomRepository;
```

Comment:
this is just a nitpick, but the order of variables should be same as in the constructor 
```suggestion
    private readonly IRoomRepository _roomRepository;

    private readonly ITrackerRepository _trackerRepository;
```